File: src-tauri/src/responses.rs
use serde::Deserialize;

#[derive(Deserialize)]
pub struct NormalResponse {
    pub success: bool,
}

#[derive(Deserialize)]
pub struct PowerResponse {
    pub on: bool,
}

#[derive(Deserialize)]
pub struct InfoResponse {
    pub ip: String,
}

File: src/lib/types/responses.ts
export type Color = [number, number, number];

export type WifiData = {
    bssid: string;
    rssi: number;
    signal: number;
    channel: number;
};

export type Segments = {
    start: number;
    stop: number;
    length: number;
    col: Color[];
};

export type StateResponse = {
    on: boolean;
    bri: number;
    ps: number;
    seg: Segments[];
};

export type InfoResponse = {
    ver: string;
    name: string;
    arch: string;
    core: string;
    freeheap: number;
    uptime: number;
    opt: number;
    brand: string;
    product: string;
    mac: string;
    ip: string;
    wifi: WifiData;
};

export type NormalResponse = {
    success: boolean;
};

export type PowerResponse = {
    on: boolean;
};
